src/reducers/resourceHelpers.js

```diff
@@ -202,6 +205,53 @@ export const updateValueErrors = (state, valueKey) => {
     if (value.uri && !isValidURI(value.uri)) errors.push("Invalid URI")
   }
 
+  if (
+    propertyTemplate.type === "literal" &&
+    propertyTemplate.validationDataType ===
+      "http://www.w3.org/2001/XMLSchema/integer" &&
+    Number.isNaN(parseInt(value.literal, 10))
+  )
+    errors.push(
+      `Literal validationDataType is 'http://www.w3.org/2001/XMLSchema/integer' but '${value.literal}' is not an integer`
```

Comment:
Would prefer `Expected datatype is ...` rather than `Literal validationDataType ...` here and below.

Reply:
now done.